File: arrays/readability.cpp
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main()
{
    string text = get_string("Text: ");

    int ln = strlen(text);

    int i, grade;
    float lcount=0, wcount=1, scount=0, L, S;

    for (i = 0; i < ln; i = i + 1){
        if (isalpha(text[i]))
        {
            lcount = lcount + 1;
        }

        if (text[i] == ' ')
        {
            wcount = wcount + 1;
        }

        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            scount = scount + 1;
        }
    }

    // printf("%d letter(s)\n", lcount);
    // printf("%d word(s)\n", wcount);
    // printf("%d sentence(s)\n", scount);

    L = round((lcount * 100) / wcount);
    S = round((scount * 100) / wcount);

    grade = 0.0588 * L - 0.296 * S - 15.8;

    printf("Grade %d\n", grade);

}

File: python/readability/readability.py
text = input("Text: ")

letters = 0
words = 1
sentences = 0

ln = len(text)

for i in range(ln):
    if text[i].isalpha():
        letters = letters + 1

    if text[i] == ' ':
        words = words + 1

    if text[i] == '.' or text[i] == '?' or text[i] == '!':
        sentences = sentences + 1

L = letters * 100 / words
S = sentences * 100 / words

grade = round(0.0588 * L - 0.296 * S - 15.8)

if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print("Grade ", grade)

